### PR TITLE
 Move hello world to main repo 

### PR DESCRIPTION
**Pull request by jamestouri**
https://github.com/medplum/medplum/issues/1401

Tested with a separate repo and added examples/james-hello-world 

Screenshot 2023-05-11 at 12 48 33 PM

Made 1 change to it and it was added

Screenshot 2023-05-11 at 12 56 37 PM

Changed the tsconfig.json in medplum-hello-world and got it to run 

Screenshot 2023-05-12 at 9 17 57 AM

### CODE REVIEW

File: .github/workflows/sync-repos.yml [hunk 1]
```diff
@@ -27,14 +27,14 @@ jobs:
               changed_paths+="$path;"
             fi
           done
-          echo "::set-output name=changed_paths::$changed_paths"
```
Comment (jamestouri) [inline .github/workflows/sync-repos.yml:30]: According to Github, set-output is deprecated and will be disabled soon, and advised to use GITHUB_ENV

File: examples/medplum-hello-world/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (jamestouri) [inline examples/medplum-hello-world/tsconfig.json:1]: Previous tsconfig

```
{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "lib": ["DOM", "DOM.Iterable", "ESNext"],
    "allowJs": false,
    "skipLibCheck": false,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react"
  },
  "include": ["src"]
}
```

File: .github/workflows/sync-repos.yml [hunk 3]
```diff
@@ -63,7 +63,7 @@ jobs:
             git clone "$repo"
 
             # Copy changed files to the listening repo
-            rsync -a --delete --exclude .git/ "${initial_working_directory}/${path%}"/ ${folder_name}/
+            rsync -a --delete --exclude .git/ --exclude tsconfig.json "${initial_working_directory}/${path%}"/ ${folder_name}/
```
Comment (jamestouri) [inline .github/workflows/sync-repos.yml:66]: Ignoring tsconfig.json

File: examples/medplum-hello-world/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  "private": true,
+  "version": "0.0.0",
+  "scripts": {
+    "dev": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview"
+  },
+  "devDependencies": {
+    "@emotion/react": "11.10.8",
+    "@mantine/core": "6.0.9",
```
Comment (codyebberson) [inline examples/medplum-hello-world/package.json:12]: These version numbers are out of sync w/ mantine in `app` and `react`.

Can we manually sync version numbers for now?  Or we can update the `scripts/upgrade.sh` tool to include `examples`, and do an automated update.

It's important that we keep version numbers in sync to maintain our sanity.

File: examples/medplum-hello-world/package.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    "@mantine/hooks": "6.0.9",
+    "@mantine/notifications": "6.0.9",
+    "@medplum/core": "2.0.17",
+    "@medplum/fhirtypes": "2.0.17",
+    "@medplum/react": "2.0.17",
+    "@tabler/icons-react": "2.17.0",
+    "@types/node": "^18.16.2",
```
Comment (codyebberson) [inline examples/medplum-hello-world/package.json:19]: For `dependencies` and `devDependencies`, we use exact version numbers, so we can remove the "^" here.

File: examples/medplum-hello-world/LICENSE.txt
Comment (codyebberson) [inline examples/medplum-hello-world/LICENSE.txt:1]: Can we remove this, and copy over the root LICENSE.txt?

**Review by codyebberson — COMMENTED** [review]
Nice, thanks @jamestouri , this is a big meaty one.

Main comments to look at are comments on `package-lock.json` and `LICENSE.txt`

We should align the version numbers too

We may want to add `examples/*` to `scripts/reinstall.sh` and `scripts/upgrade.sh`

File: .github/workflows/sync-repos.yml [hunk 6]
```diff
@@ -63,7 +63,8 @@ jobs:
...
+            cp LICENSE.txt ${folder_name}/
```
Comment (jamestouri) [inline .github/workflows/sync-repos.yml:67]: cp the LICENSE.txt to the repo

**Review by codyebberson — APPROVED** [review]

**Pull request merged**